pkg/declared/resources.go

```diff
@@ -45,7 +45,7 @@ type Resources struct {
 }
 
 // Update performs an atomic update on the resource declaration set.
-func (r *Resources) Update(ctx context.Context, objects []client.Object, commit string) ([]client.Object, status.Error) {
+func (r *Resources) Update(ctx context.Context, objects []client.Object, commit string, oldCommit string) ([]client.Object, status.Error) {
```

Comment:
What if we store the commit in the Resources? That way when Update is called we have the old commit on the struct and the new commit coming in as an arg.

Reply:
How to know if a commit is the first one? Will the field be empty when a Resource is first created

Reply:
Empty string is the default value for any string field.

Reply:
Done! Added a separate field in Resources.